src/gui/Connected.qml

```diff
@@ -82,13 +93,99 @@ Item {
         anchors.verticalCenter: headphones.verticalCenter
         color: textColour
     }
-    
+
+    Text {
+        id: networkStatsHeader
+        x: 120 * virtualstudio.uiScale
+        text: "Network"
+        font {family: "Poppins"; pixelSize: fontMedium * virtualstudio.fontScale * virtualstudio.uiScale }
+        anchors.verticalCenter: network.verticalCenter
+        color: textColour
+    }
+
+    function getNetworkStatsText (networkStats) {
+        let packetsSent = networkStats.packetsSent;
+        let packetsReceived = networkStats.packetsReceived;
+        let minRtt = networkStats.minRtt;
+        let maxRtt = networkStats.maxRtt;
+        let avgRtt = networkStats.avgRtt;
+
+        let texts = ["Measuring stats ...", "", ""];
+
+        if (!minRtt || !maxRtt) {
+            return texts;
+        }
+
+        texts[0] = "<b>" + minRtt + " ms - " + maxRtt + " ms</b>, avg " + avgRtt + " ms round-trip time";
+        texts[1] = (100 - (packetsReceived * 100 / packetsSent)) + "% packet loss";
+
+        let quality = "poor";
+        if (avgRtt <= 25) {
+
+            if (maxRtt <= 30 && packetsSent === packetsReceived) {
+                quality = "excellent";
+            } else {
+                quality = "good";
+            }
+
+        } else if (avgRtt <= 30) {
+
+            if (packetsSent === packetsReceived) {
+                quality = "good";
+            } else {
+                quality = "fair";
+            }
+
+        } else if (avgRtt <= 35) {
+
+            if (packetsSent === packetsReceived) {
+                quality = "fair";
+            } else {
+                quality = "low";
+            }
+
+        }
+
+        texts[2] = "Your connection quality is <b>" + quality + "</b>."
+        return texts;
+    }
+
+    Text {
+        id: netstat0
+        text: getNetworkStatsText(virtualstudio.networkStats)[0]
+        font {family: "Poppins"; pixelSize: fontSmall * virtualstudio.fontScale * virtualstudio.uiScale }
+        topPadding: smallTextPadding
+        anchors.left: inputDeviceHeader.left
+        anchors.top: networkStatsHeader.bottom
+        color: textColour
+    }
+
+    Text {
+        id: netstat1
+        text: getNetworkStatsText(virtualstudio.networkStats)[1]
+        font {family: "Poppins"; pixelSize: fontSmall * virtualstudio.fontScale * virtualstudio.uiScale }
+        topPadding: smallTextPadding
+        anchors.left: inputDeviceHeader.left
+        anchors.top: netstat0.bottom
+        color: textColour
+    }
+
+    Text {
+        id: netstat2
+        text: getNetworkStatsText(virtualstudio.networkStats)[2]
+        font {family: "Poppins"; pixelSize: fontSmall * virtualstudio.fontScale * virtualstudio.uiScale }
+        topPadding: smallTextPadding
+        anchors.left: inputDeviceHeader.left
+        anchors.top: netstat1.bottom
+        color: textColour
+    }
     //43 822
```

Comment:
not sure what this is for, maybe delete?

Reply:
these anchors are definitely needed so that we can stack each line of text on top of each other, this is done by anchoring the top of one line to the bottom of the previous

Reply:
Sorry I meant this `//43 822` line

Reply:
oh yeah i also don't know what that's for, i took it out